pkg/export/export.go

```diff
@@ -225,9 +228,20 @@ func (alwaysLease) OnLeaderChange(f func()) {
 	// We never lose the lease as it's always owned.
 }
 
+func formatUserAgent(ua, env string) string {
+	re := regexp.MustCompile(`(.*\()(mode:.*\))$`)
+	if re.MatchString(ua) {
+		str := fmt.Sprintf("${1}env:%s;${2}", env)
+		return re.ReplaceAllString(ua, str)
+	} else {
+		return ua + fmt.Sprintf(" (env:%s)", env)
+	}
+}
+
 func newMetricClient(ctx context.Context, opts ExporterOpts) (*monitoring.MetricClient, error) {
 	// Identity User Agent for all gRPC requests.
-	ua := strings.TrimSpace(fmt.Sprintf("%s/%s %s", ClientName, Version, opts.UserAgent))
```

Comment:
Again - the main idea is to prevent overwriting the actual export library and version that was used to send the data, which is why we use a suffix.

In practice, I'm sure this field is rarely used or overwritten by most people. However, it's nice knowing that some aspects of the user agent are built-in to the binary sending the data over. It also ensures the library complies with the [HTTP semantics RFC](https://www.rfc-editor.org/rfc/rfc7231#section-5.5.3) around user-agent formatting in at least one respect.

Happy to discuss more though. I do agree there's some confusion. But I'm leaning towards either:
- Just update the flag help text, or
- Fully rename the flag to `export.user-agent-suffix`

Reply:
It's not a good idea that an extra interface exposing out there. This flag is used in at least our Cloud Doc. I don't have data whether it get overwritten by people; Even though it is not overwritten by most people, I believe the intention is to not get this variable overwritten by users.

 A built time variable should still be preferred so users won't temper it. See an early comment in the evaluator too: https://github.com/GoogleCloudPlatform/prometheus-engine/blob/092fb1241fc6c83b39292225e99abe67fcd70829/cmd/rule-evaluator/main.go#L83-L85 

I won't reject this PR as it is on par with the existing surface. 

but I strongly suggest
- deprecate this flag (even better, marking this flag DEPRECATED in this PR), and file an issue about deprecating this flag, if not created yet - as if we want to make sure this flag is not used or overwritten by other people. If we don't want to get confused that what the alternative is, we mark this flag DEPRECATED after we can ingest built-time variable and has a separate `mode` variable. 
- For the use case that we want to allow setting `mode`, only allow specific ENUM or STRING to be set, otherwise the mode won't be set in user agent.